datamodel/create_roles.py

```diff
@@ -0,0 +1,237 @@
+#!/usr/bin/env python3
+
+import re
+from argparse import ArgumentParser, BooleanOptionalAction
+from typing import Optional
+
+from pgserviceparser import service_config
+
+try:
+    import psycopg
+except ImportError:
+    import psycopg2 as psycopg
+
+
+def get_db_identifier(pg_service: str, modelname: str):
+    dbname = service_config(pg_service).get("dbname")
+    db_identifier = re.sub(f"{modelname}_|teksi_", "", dbname)
+    return db_identifier
+
+
+def create_role(pg_service: str, role: str, in_role: str = None):
+    conn = psycopg.connect(f"service={pg_service}")
+    cur = conn.cursor()
+    cur.execute(f"SELECT 1 FROM pg_roles WHERE rolname='{role}'")
+    role_exists = cur.fetchone()
+    if not role_exists:
+        role_sql = f"""
+            CREATE ROLE {role} NOSUPERUSER INHERIT NOCREATEDB NOCREATEROLE NOREPLICATION;
+        """
+        print(f"created role {role}")
+        cur.execute(role_sql)
+        cur.execute(f"SELECT 1 FROM pg_roles WHERE rolname='{in_role}'")
+        inrole_exists = cur.fetchone()
+        if inrole_exists:
+            in_role_sql = f"""
+                GRANT {in_role} TO {role};
+            """
+            cur.execute(in_role_sql)
+            print(f"granted {in_role} to role {role}")
+    conn.commit()
+    conn.close()
+
+
+def grant_privileges(pg_service: str, modelname: str, roles: list, ext_schema: str = None):
```

Comment:
it doesn't sound super robust to have the roles as a list and use it by indexes.
I would either use a dict or specific arguments for each role.